File: map2d/src/main/java/net/ponderingprogrammer/map2d/Cell.java
package net.ponderingprogrammer.map2d;

public class Cell<T> {
    private Coord coord;
    private T value;

    public Cell(Coord coord, T value) {
        this.coord = coord;
        this.value = value;
    }

    public Coord getCoord() {
        return coord;
    }

    public T getValue() {
        return value;
    }

    public void setValue(T value) {
        this.value = value;
    }
}

File: NProcGen/src/PonderingProgrammer.Map2d/Cell.cs
namespace PonderingProgrammer.Map2d
{
    public class Cell<T>
    {
        public Coord Coord { get; }
        public T Value { get; set; }

        public Cell(Coord coord, T value)
        {
            Coord = coord;
            Value = value;
        }
    }
}
